client/src/test/java/model/DataProvider.java

```diff
@@ -24,6 +23,33 @@
     return fields;
   }
 
+  @org.testng.annotations.DataProvider(name = "fieldsOnBoundaries")
+  public static Object[] fieldsOnBoundaries() {
+
+    return  new Object[][]{ {0, 2}, { 0, 5}, {0, 7},
+              {9, 1}, { 9, 3}, {9, 6},
+              {2, 0}, { 4, 0}, {6, 0},
+              {1, 9}, { 3, 9}, {8, 9} };
```

Comment:
formatting - spaces before some numbers - maybe you want to change it

Reply:
ok